bundle/bind.go

```diff
@@ -37,24 +44,49 @@ func (e *executor) Bind(
 
 	go func() {
 		e.actionStarted()
-		executionContext, err := e.executeApb(
-			"bind", instance.Spec, instance.Context, parameters)
+		// Create namespace name that will be used to generate a name.
+		ns := fmt.Sprintf(fmt.Sprintf("%s-%.4s-", instance.Spec.FQName, bindAction))
+		// Create the podname
+		pn := fmt.Sprintf("bundle-%s", uuid.New())
+		targets := []string{instance.Context.Namespace}
+		labels := map[string]string{
+			"bundle-fqname":   instance.Spec.FQName,
+			"bundle-action":   bindAction,
+			"bundle-pod-name": pn,
+		}
+		ec := runtime.ExecutionContext{
+			BundleName: pn,
+			Targets:    targets,
+			Metadata:   labels,
+			Action:     bindAction,
+			Image:      instance.Spec.Image,
+		}
+
+		serviceAccount, namespace, err := runtime.Provider.CreateSandbox(pn, ns, targets, clusterConfig.SandboxRole, labels)
+		ec.Account = serviceAccount
+		ec.Location = namespace
```

Comment:
The order of this initialization feels weird. We create the `ExecutionContext` (partially), then get 2 values, which we add to the context. Feels like we should get the `serviceAccount` and the `namespace` first, handle the error. If it's okay, then create the ExecutionContext.